error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ComparatorRules.java

```diff
@@ -92,6 +95,24 @@ Comparator<T> after(Comparator<T> cmp) {
     }
   }
 
+  /** Don't explicitly compare enums by their ordinal. */
+  abstract static class ComparingEnum<E extends Enum<E>, T> {
+    @Placeholder(allowsIdentity = true)
+    abstract E toEnumFunction(@MayOptionallyUse T value);
+
+    @BeforeTemplate
+    @SuppressWarnings("EnumOrdinal" /* This violation will be rewritten. */)
+    Comparator<T> before() {
+      return comparingInt(v -> toEnumFunction(v).ordinal());
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    Comparator<T> after() {
+      return comparing(v -> toEnumFunction(v));
+    }
+  }
```

Comment:
This will rewrite `comparingInt(e -> e.getKey().ordinal())` to `comparing(v -> v.getKey())`. That result will not also be rewritten to `comparingByKey()`, because [MapEntryComparingByKey](https://github.com/PicnicSupermarket/error-prone-support/blob/bb2b1e60349b546cab99bab54bc66e0dbce1a122/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/MapEntryRules.java#L47-L60) matches against `Map.Entry::getKey` but not the associated lambda expression. This limitation applies to many of our rules, so fixing it is out of scope. (We should really "just" fix the `MethodReferenceUsage` check.)

I'm mentioning this because it'll prevent a cleanup opportunity in our internal code base, I see.

Reply:
Or combine this with OpenRewrite's https://docs.openrewrite.org/recipes/staticanalysis/replacelambdawithmethodreference if we can...